kcidb/cache/__init__.py

```diff
@@ -0,0 +1,149 @@
+import hashlib
+import requests
+from email.header import Header
+import urllib.parse
+from google.cloud import storage
+
+class Client:
+    def __init__(self, bucket_name, max_store_size):
+        """
+        Initialize a cache client.
+        Args:
+            bucket_name:    The name of the GCS bucket containing the cache.
+            max_store_size: Maximum size the file can have to be stored.
+        """
+        self.bucket_name = bucket_name
+        self.max_store_size = max_store_size
+        self.client = storage.Client()
+
+    def _extract_content_disposition(self, response):
+        content_disposition = response.headers.get("Content-Disposition")
+
+        if content_disposition is None:
+            parsed_url = urllib.parse.urlparse(response.url)
+            original_filename = parsed_url.path.rsplit('/', 1)[-1]
+
+            encoded_filename = Header(original_filename, "utf-8").encode()
+
+            content_disposition = f'inline; filename="{encoded_filename}"'
+
+        return content_disposition
+
+    def store(self, url):
+        """
+        Attempt to store a URL in the cache. The URL contents is not
+        downloaded if it's already in the cache or if it doesn't match the
+        requirements (max_store_size).
+        Args:
+            url:    The URL to try to cache.
+        """
+        object_name = self._format_object_name(url)
+        blob = self.client.bucket(self.bucket_name).blob(object_name)
+
+        if blob.exists():
+            print(f"URL '{url}' is already cached.")
```

Comment:
Please remove printouts here and everywhere else in this module. These are not suitable for a library. Instead, if needed, we should provide a way for the caller to find out what exactly happened (e.g. through a return value) and make a decision what to do with that, whether to `print` something, to log a message, or do something else.

Reply:
okay